File: src/test/java/com/agiledeveloper/WordCountTest.java
package com.agiledeveloper;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.io.FileNotFoundException;

import org.junit.jupiter.api.Test;

public class WordCountTest {
  @Test
  void noFile() {
    Exception ex = assertThrows(RuntimeException.class, () -> WordCount.count("nofile", ""));
    
    assertTrue(ex.getCause() instanceof FileNotFoundException);
  } 
  
  @Test
  void wordCount() {
    assertAll(
      () -> assertEquals(0, WordCount.count("file1.txt", "lambda"))
      , () -> assertEquals(1, WordCount.count("file2.txt", "lambda"))
      , () -> assertEquals(3, WordCount.count("file3.txt", "lambda"))
      , () -> assertEquals(0, WordCount.count("WordCount.java", "lambda"))
      , () -> assertEquals(5, WordCount.count("WordCountTest.java", "lambda"))
    );
  }
}

File: src/test/scala/com/agiledeveloper/WordCountTest.scala
package com.agiledeveloper

import java.io._
import org.scalatest._

class WordCountTest extends WordSpec with Matchers {

  "The WordCount" should {
    "fail when given a non-existing file" in {
      val error = the [RuntimeException] thrownBy {
        WordCount.count("nofile", "")
      }
      error.getCause shouldBe a [FileNotFoundException]
    }

    "correctly cound words" in {
      WordCount.count("file1.txt", "lambda") shouldBe 0
      WordCount.count("file2.txt", "lambda") shouldBe 1
      WordCount.count("file3.txt", "lambda") shouldBe 3
      WordCount.count("WordCount.java", "lambda") shouldBe 0
      WordCount.count("WordCountTest.java", "lambda") shouldBe 5
    }
  }
}

